src/compiler/checker.ts

```diff
@@ -21909,11 +21909,13 @@ namespace ts {
                 // check private/protected variable access
                 const parent = node.parent.parent;
                 const parentType = getTypeForBindingElementParent(parent);
-                const name = node.propertyName || <Identifier>node.name;
-                const property = getPropertyOfType(parentType, getTextOfPropertyName(name));
-                markPropertyAsReferenced(property, /*nodeForCheckWriteOnly*/ undefined, /*isThisAccess*/ false); // A destructuring is never a write-only reference.
-                if (parent.initializer && property) {
-                    checkPropertyAccessibility(parent, parent.initializer, parentType, property);
+                const name = node.propertyName || node.name;
+                if (!isBindingPattern(name)) {

```

Comment:
Is there a documented bug this change fixes? If not, can you come up with a test case which would have crashed/error'd/behaved differently before? It'd be nice to just the coverage if possible.

Reply:
Previously `getTextOfPropertyName` would have just returned `undefined`, then so would `getPropertyOfType` and it would be ignored. So there's no observable difference in behavior.